manageiq-operator/pkg/helpers/miq-components/memcached.go

```diff
@@ -101,19 +101,18 @@ func NewMemcachedService(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*cor
 			Name:      "memcached",
 			Namespace: cr.ObjectMeta.Namespace,
 		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{},
+		},
 	}
 
 	f := func() error {
 		if err := controllerutil.SetControllerReference(cr, service, scheme); err != nil {
 			return err
 		}
 		addAppLabel(cr.Spec.AppName, &service.ObjectMeta)
-		service.Spec.Ports = []corev1.ServicePort{
-			corev1.ServicePort{
-				Name: "memcached",
-				Port: 11211,
-			},
-		}
+		service.Spec.Ports[0].Name = "memcached"
```

Comment:
I'm worried that his `Ports` array will come back as `nil` if there isn't anything set. Can you check for nil here?

Reply:
Also it seems like it wouldn't be worth setting it in the service spec above, right?

Reply:
Nope, turns out I don't need that

Reply:
Fixed the nil issue and tested that it gets recreated is the service is deleted